apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -197,6 +199,9 @@ export function OAuth2Form() {
         // done!  redirect with access code.
         setShowPendingRedirectUI(true)
         const redirectUri = queryParams.get('redirect_uri')!
+        cookies.remove(
+          SynapseConstants.ACCOUNT_SITE_PROMPTED_FOR_LOGIN_COOKIE_KEY,
```

Comment:
Cleanup.  Just before redirecting, also remove the "account site prompted" cookie state